client/src/main/java/org/apache/crail/metadata/DataNodeStatistics.java

```diff
@@ -22,25 +22,29 @@
 import java.nio.ByteBuffer;
 
 public class DataNodeStatistics {
-	public static final int CSIZE = 12;
+	public static final int CSIZE = 14;
 	
 	private long serviceId;
 	private int freeBlockCount;
+	private short status;
```

Comment:
Can you provide possible values for the status? If I understand it correctly it uses RpcErrors.ERR_DATANODE_STOP only? Error seems not to fit to status. Maybe we should define a new class DataNodeStatus where we define the possible status?

Reply:
Yes, that is correct. Right now the status field is only changed to RpcErrors.ERR_DATANODE_STOP when a datanode is selected for shutdown. Replacing it with a new class seems like a good idea.